### PR TITLE
close #14 vendor multi images

### LINKED ISSUE #14: Upload multi images

### PR DESCRIPTION
**Pull request by theachoem**
# Vendor Multi-images
- Add /vendor_photos views inside admin/vendors
- Crud multiple images
- Response photos in vendor API:
  - api/v2/storefront/vendors?include=photos
  - api/v2/storefront/vendors/:slug?include=photos

## Demo
|  |
| - |
|  |
| |
|  |

### CODE REVIEW

File: app/models/spree_cm_commissioner/vendor_photo.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        {
+          url: polymorphic_path(attachment.variant(resize: size), only_path: true),
+          width: width,
+          height: height
+        }
+      end
+    end
+
+    def mobile_styles
+      {
```
Comment (theachoem) [inline app/models/spree_cm_commissioner/vendor_photo.rb:20]: I'm not sure how many image sizes we should choose. So I put all of them. Anyways, we also include original size in serializer.

File: app/models/spree_cm_commissioner/vendor_photo.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        plp_and_carousel_sm: "350x468>",
+        plp_and_carousel_md: "222x297>",
+        plp_and_carousel_lg: "278x371>",
+        large: "600x600>",
+        plp: "278x371>",
+        zoomed: "650x870>"
+      }
+    end
+  end
+end
```
Comment (theachoem) [inline app/models/spree_cm_commissioner/vendor_photo.rb:36]: I'm not sure how many image sizes we should choose. So I put all of them. Anyways, we also include original size in serializer.

File: app/models/spree_cm_commissioner/vendor_photo.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        plp_and_carousel: "448x600>",
+        plp_and_carousel_xs: "254x340>",
+        plp_and_carousel_sm: "350x468>",
+        plp_and_carousel_md: "222x297>",
+        plp_and_carousel_lg: "278x371>",
+        large: "600x600>",
+        plp: "278x371>",
+        zoomed: "650x870>"
+      }
+    end
```
Comment (theachoem) [inline app/models/spree_cm_commissioner/vendor_photo.rb:34]: I'm not sure how many image sizes we should choose. So I put all of them. Anyways, we also include original size in serializer.

File: app/models/spree_cm_commissioner/vendor_photo.rb (hunk 2 above)
Comment (theachoem) [inline app/models/spree_cm_commissioner/vendor_photo.rb:36]: I'm not sure how many image sizes we should choose. So I put all of them. Anyways, we also include original size in serializer.

File: spec/controllers/spree/api/v2/storefront/vendors_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+RSpec.describe Spree::Api::V2::Storefront::VendorsController, type: :controller do
```
Comment (kimsrung) [inline spec/controllers/spree/api/v2/storefront/vendors_controller_spec.rb:3]: we could do this in request spec.

File: app/models/spree_cm_commissioner/vendor_photo.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+          url: polymorphic_path(attachment.variant(resize: size), only_path: true),
+          width: width,
+          height: height
+        }
+      end
+    end
+
+    def mobile_styles
+      {
+        mini: "48x48>",
```
Comment (channainfo) [inline app/models/spree_cm_commissioner/vendor_photo.rb:21]: keep the image aspect  ratio the same( 4:3 ) uses these 3 sizes:
160x120
480x360
960x720

**Review by channainfo — CHANGES_REQUESTED** [review]

File: spec/serializers/spree/v2/storefront/vendor_serializer_spec.rb [hunk 6]
```diff
@@ -3,15 +3,31 @@
 RSpec.describe Spree::V2::Storefront::VendorSerializer, type: :serializer do
   describe '.serializable_hash' do
     let(:vendor) {
```
Comment (kimsrung) [inline spec/serializers/spree/v2/storefront/vendor_serializer_spec.rb:5]: to make this easy to read you could do 
```
let!(:vendor_logo) { create(:vendor_logo) }
let!(:vendor_photo1) { create(:vendor_photo) }
let!(:vendor) { create(:active_vendor, name: 'vendor', logo: vendor_logo, photos: [vendor_photo1]) }
```
... etc

File: spec/serializers/spree/v2/storefront/vendor_serializer_spec.rb [hunk 7]
```diff
@@ -3,15 +3,31 @@
...
+
       vendor.save!
       vendor.reload
+
+      vendor
     }
 
-    subject { described_class.new(vendor).serializable_hash }
+    subject {
+      described_class.new(vendor, include: [
```
Comment (kimsrung) [inline spec/serializers/spree/v2/storefront/vendor_serializer_spec.rb:23]: create 2 different context for this
1. is for serializable_hash
2. is for serializable_hash with included

Comment (theachoem) [reply spec/serializers/spree/v2/storefront/vendor_serializer_spec.rb]: Will add this too

**Pull request merged**